File: apps/web/src/app/garden/garden-strings.ts
/**
 * Every user-visible garden string, centralized so the web garden forms never
 * inline copy — see `docs/terminology.md` § Garden for the binding glossary
 * this implements. Apple's mirror lives in
 * `apps/apple/App/Shared/Garden/GardenStrings.swift`.
 *
 * The entry-kind vocabulary ("Note" / "Harvest" / "Move") is *not* duplicated
 * here: `gardenEntryKindLabel` in `./garden-photos` is already the one
 * canonical source (shared with the journal's date · kind line), so entry
 * forms import that instead.
 */
/** Shared between `location.addTitle` (dialog title) and `home.addLocation` (trigger). */
const addGrowingAreaTitle = "Add growing area";

export const gardenStrings = {
  common: {
    notesField: "Notes",
    saveLabel: "Save",
    retry: "Retry",
  },
  location: {
    kindLabel: {
      bed: "Raised bed",
      tray: "Seed tray",
      other: "Other",
    } as const,
    addTitle: addGrowingAreaTitle,
    editTitle: "Edit growing area",
    nameField: "Name",
    kindField: "Growing area",
    conditionsField: "Growing conditions",
    existingPickerCaption: "Use an existing location",
    existingPickerPlaceholder: "Search existing locations…",
    parentField: "Parent location (optional)",
    wholeArea: "Whole area",
    noLocationYet: "No location yet",
  },
  planting: {
    addTitle: "Add planting",
    editTitle: "Edit planting",
    verbs: {
      start: "Start planting",
      move: "Move everything",
      split: "Move some seedlings",
      finish: "Finish planting",
    },
    submitAdd: "Add planting",
    cropField: "Crop",
    rememberSourceLabel: "Remember that this product grows this crop",
    locationField: "Location",
    intendedDestinationField: "Intended destination",
    stateField: "Planting state",
    stateGrowing: "Growing now",
    statePlanned: "Planned",
    stateFinished: "Finished",
    sourceField: "Seed packet or plant (optional)",
    varietyField: "Variety",
    quantityField: "Approximate quantity",
    quantityPlaceholder: "A few seedlings",
    plannedWindowField: "Planned window",
    plannedWindowPlaceholder: "Early autumn",
    plannedDateField: "Planned date",
    sowedOnField: "Sowed on",
    transplantedOnField: "Transplanted on",
    inLocationSinceField: "In this location since",
    inLocationSinceHelp:
      "Optional. This confirms which older bed photos belong in the journal. Leave blank to record presence from today without guessing an earlier date.",
    datesDetailsSummary: "Dates and other details (optional)",
    datesHelp: "Leave dates blank when you don’t know them.",
    guideSummary: "Local planting guide",
    guideFieldLabel: "Guide for this ingredient",
    guideNoneOption: "No guide linked",
    startMethodField: "How are you starting?",
    startMethodSow: "Sowing seeds",
    startMethodTransplant: "Planting a seedling or plant",
    startMethodExisting: "Already growing; date unknown",
    splitQuantityField: "Quantity being moved (optional)",
    splitExplanation:
      "Remaining seedlings stay in the original location. The new planting keeps the seed source and sowing history.",
    finishExplanation:
      "This finishes only this planting. Its photos and harvest history stay available.",
    dateField: "Date",
    startingLocationField: "Location",
    destinationField: "Destination",
  },
  entry: {
    kindField: "Entry type",
    dateField: "Date",
    harvestDateField: "Harvest date",
    harvestAmountField: "Harvest amount",
    harvestAmountPlaceholder: "A handful, 6 tomatoes, 300 g…",
    locationField: "Location",
    aboutField: "About",
    aboutHelpWithPlanting:
      "This entry stays in this planting’s journal, even after it moves.",
    aboutHelpWithoutPlanting:
      "Shared with plantings known to be here on the observation date. Unknown earlier dates are not assumed.",
    moveLockedHint:
      "Correct move dates in the planting’s location history so its journal stays consistent.",
    loadFailed: "Could not load locations and plantings.",
  },
  home: {
    addPlanting: "Add planting",
    addLocation: addGrowingAreaTitle,
    finishSelected: (n: number) => `Finish selected (${n})`,
    allEntries: "All entries",
    emptyTitle: "Start with what’s growing today",
    emptyBody:
      "Add a bed, tray, or growing area, then record your crops. Leave unknown dates blank.",
    noPlantingsInLocation: "No current or planned plantings.",
    logEntry: "Log entry",
    editConditions: "Edit growing area",
    bedJournal: "Area journal",
    finishedPlantingsSummary: (n: number) => `Finished plantings (${n})`,
    finishedEmpty: "Finished plantings will stay here with their history.",
    finishDialogTitle: "Finish selected plantings",
    finishSubmit: (label: string) => `Finish ${label}`,
    finishPartialReport: (finished: string, total: string) =>
      `Finished ${finished} of ${total}.`,
    loadFailed: "Could not load the garden",
  },
  photos: {
    entryHelp:
      "Keep the whole area or add a close-up. Photos save with this entry.",
    plantingHelp:
      "Photos of this planting today. They save as its first journal entry.",
  },
} as const;

File: apps/apple/App/Shared/Garden/GardenStrings.swift
import Foundation

/// The garden glossary as one source of truth. `docs/terminology.md` § Garden is the spec; every
/// garden view reads copy from here instead of inlining it, so the two clients (this file and
/// `apps/web/src/app/garden/garden-strings.ts`) cannot drift from each other or from the doc.
enum GardenStrings {
    // MARK: - Growing areas (locations)

    static let addGrowingArea = "Add growing area"
    static let noLocationYet = "No location yet"
    static let wholeArea = "Whole area"
    static let location = "Location"

    // MARK: - Plantings

    static let addPlanting = "Add planting"
    static let startPlanting = "Start planting"
    static let moveEverything = "Move everything"
    static let moveSomeSeedlings = "Move some seedlings"
    static let finishPlanting = "Finish planting"
    static let quantity = "Quantity"
    static let approximateQuantityPrompt = "Approximate, e.g. \u{22}6 plants\u{22} or \u{22}2 lb\u{22}"
    static let about = "About"

    /// The verb-plus-explanation copy shown when starting a garden verb, matching web's
    /// `plantingActionLabels`/explanation copy.
    static let moveEverythingExplanation =
        "Every seedling in this planting moves to the new location. The old location is recorded as this planting's history."
    static let moveSomeSeedlingsExplanation =
        "Some seedlings move to a new location as their own planting; the rest stay where they are."
    static let finishPlantingExplanation =
        "Finishing records the last day this planting was active. It can still be logged against afterward."
    static let startPlantingExplanation =
        "Starting records where and how this planting entered the garden."

    // MARK: - Entries / journal

    static let logEntry = "Log entry"
    static let save = "Save"
    static let cancel = "Cancel"
    static let notes = "Notes"
    static let notesPlaceholder = "Anything useful to remember"
    static let gardenJournal = "Garden journal"

    // MARK: - Dates

    static let date = "Date"
    static let inThisLocationSince = "In this location since"
    static let confirmLocationDates = "Confirm location dates"
    static let correctLocationDates = "Correct location dates"
    static let recordedAsActual = "Actual date"
    static let recordedAsLater = "Recorded later"
    static let earlierPresenceUnknown =
        "Earlier presence is unknown. Add a date you know to include older bed photos."

    // MARK: - Guides

    static let viewSource = "View source"
    static let reviewed = "Reviewed"
    static let publishedOrRevised = "Published/revised"
    static let basedOn = "Based on"
    static let sourcesMayDiffer =
        "Sources may differ. Use the windows as local context; your bed's conditions still matter."

    // MARK: - Empty / error states

    static let noGardenLocationsYet = "No garden locations yet"
    static let noGardenLocationsDescription = "Add a growing area, then add what is growing."
    static let couldNotLoadGarden = "Couldn't load the garden"
    static let retry = "Retry"
}
